### PR TITLE
Upgrade to webpack 3

### LINKED ISSUE #201: Upgrade to Webpack 3

### PR DESCRIPTION
**Pull request by zamotany**
Closes #201

### CODE REVIEW

File: src/utils/makeReactNativeConfig.js [hunk 1]
```diff
@@ -137,23 +148,6 @@ const getDefaultConfig = ({
     /**
```
Comment (satya164) [inline src/utils/makeReactNativeConfig.js]: Maybe wanna remove this comment too?

File: src/utils/makeReactNativeConfig.js [hunk 2]
```diff
@@ -82,10 +81,22 @@ const getDefaultConfig = ({
...
-        },
+        use: [
+          {
+            loader: require.resolve('thread-loader'),
+          },
+          {
+            loader: require.resolve('babel-loader'),
+            options: Object.assign({}, getBabelConfig(root), {
+              /**
+             * This enables caching results in ./node_modules/.cache/babel-loader/
```
Comment (satya164) [inline src/utils/makeReactNativeConfig.js]: Comment indentation is wrong

File: src/utils/makeReactNativeConfig.js [hunk 3]
```diff
@@ -137,23 +148,6 @@ const getDefaultConfig = ({
     /**
      * HappyPack runs transforms in parallel to improve build performance
      */
```
Comment (thymikee) [inline src/utils/makeReactNativeConfig.js]: leftover comment?

File: src/messages/bundleBuilt.js [hunk 4]
```diff
@@ -23,7 +23,8 @@ module.exports = ({
 }) => {
   const heading = stats.hasWarnings()
     ? chalk.yellow('Built with warnings')
-    : 'Built successfully!';
+    : `Built successfully! (${(stats.toJson({ timing: true }).time /
```
Comment (satya164) [inline src/messages/bundleBuilt.js]: Maybe `Built successfully in ${time}ms!` is better

**Comment by thymikee** [comment]
btw, are there any noticeable perf improvements?

**Pull request merged**